File: src/main/java/tech/jhipster/lite/shared/github/domain/GithubRepository.java
package tech.jhipster.lite.shared.github.domain;

import java.util.List;
import tech.jhipster.lite.shared.github.domain.authentication.GithubAuthenticationCode;
import tech.jhipster.lite.shared.github.domain.authentication.GithubAuthorizationUrl;
import tech.jhipster.lite.shared.github.domain.authentication.GithubToken;

public interface GithubRepository {
  GithubAuthorizationUrl buildAuthorizationUrl();
  GithubToken authenticate(GithubAuthenticationCode code);
  List<GithubOrganization> listUserOrganizations(GithubToken token);
  List<GithubRepositoryInfo> listUserRepositories(GithubToken token);
}

File: src/main/webapp/app/module/domain/GithubRepository.ts
import type { GithubAuthorizationUrl } from '@/module/domain/GithubAuthorizationUrl';
import type { GithubOrganization } from '@/module/domain/GithubOrganization';
import type { GithubToken } from '@/module/domain/GithubToken';
import type { GithubRepositoryInfo } from './GithubRepositoryInfo';

export interface GithubRepository {
  getAuthorizationUrl(): Promise<GithubAuthorizationUrl>;
  authenticate(code: string): Promise<GithubToken>;
  listOrganizations(token: GithubToken): Promise<GithubOrganization[]>;
  authenticated(): boolean;
  listRepositories(token: GithubToken, organization: string): Promise<GithubRepositoryInfo[]>;
}
